In [1]:
import numpy as np
import pandas as pd  
import seaborn as sns
import matplotlib.pyplot as plt 

import xgboost as xgb

In [3]:
all_data_clean = pd.read_csv('all_data_clean.csv')
X = all_data_clean.drop(['%vote'], axis = 1)
y = all_data_clean.iloc[:, 1]

In [4]:
model = xgb.XGBRegressor(objective='reg:squarederror',
                        reg_alpha=0,
                        learning_rate=0.25,
                        max_depth=9,
                        n_estimators=185)

In [6]:
model.fit(X, y)

XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, device=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=0.25, max_bin=None,
             max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=9, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             multi_strategy=None, n_estimators=185, n_jobs=None,
             num_parallel_tree=None, random_state=None, ...)

In [14]:
url = 'https://www.baseball-reference.com/awards/hof_2024.shtml'
data2024 = pd.read_html(url, header = 1)[0]

In [15]:
data2024.head()

Rk            Name   YoB % of Ballots  HOFm  HOFs  Yrs   WAR  WAR7  JAWS  \
0   1     Todd Helton   6th        72.2%   175    59   17  61.8  46.6  54.2   
1   2    Billy Wagner   9th        68.1%   107    24   16  27.7  19.8  23.7   
2   3    Andruw Jones   7th        58.1%   109    34   17  62.7  46.4  54.6   
3   4  Gary Sheffield  10th        55.0%   158    61   22  60.5  38.0  49.3   
4   5  Carlos Beltrán   2nd        46.5%   126    52   20  70.1  44.4  57.3   

   ...   WHIP    G.1   GS     SV     IP    H.1  HR.1   BB.1      SO  \
0  ...    NaN    NaN  NaN    NaN    NaN    NaN   NaN    NaN     NaN   
1  ...  0.998  853.0  0.0  422.0  903.0  601.0  82.0  300.0  1196.0   
2  ...    NaN    NaN  NaN    NaN    NaN    NaN   NaN    NaN     NaN   
3  ...    NaN    NaN  NaN    NaN    NaN    NaN   NaN    NaN     NaN   
4  ...    NaN    NaN  NaN    NaN    NaN    NaN   NaN    NaN     NaN   

    Pos Summary  
0       *3H/7D9  
1            *1  
2      *89H7D/3  
3  *9*7*5*D6H/3  
4    *8*9*DH/73  

[5 rows x 38 columns]

In [16]:
data2024_clean = data2024[~data2024['Pos\xa0Summary'].str.contains('*1', regex = False)]
data2024_clean.head()

Rk            Name   YoB % of Ballots  HOFm  HOFs  Yrs    WAR  WAR7  JAWS  \
0   1     Todd Helton   6th        72.2%   175    59   17   61.8  46.6  54.2   
2   3    Andruw Jones   7th        58.1%   109    34   17   62.7  46.4  54.6   
3   4  Gary Sheffield  10th        55.0%   158    61   22   60.5  38.0  49.3   
4   5  Carlos Beltrán   2nd        46.5%   126    52   20   70.1  44.4  57.3   
5   6  Alex Rodriguez   3rd        35.7%   390    77   22  117.5  64.3  90.9   

   ...  WHIP  G.1  GS  SV  IP  H.1  HR.1  BB.1  SO   Pos Summary  
0  ...   NaN  NaN NaN NaN NaN  NaN   NaN   NaN NaN       *3H/7D9  
2  ...   NaN  NaN NaN NaN NaN  NaN   NaN   NaN NaN      *89H7D/3  
3  ...   NaN  NaN NaN NaN NaN  NaN   NaN   NaN NaN  *9*7*5*D6H/3  
4  ...   NaN  NaN NaN NaN NaN  NaN   NaN   NaN NaN    *8*9*DH/73  
5  ...   NaN  NaN NaN NaN NaN  NaN   NaN   NaN NaN      *6*5DH/3  

[5 rows x 38 columns]

In [17]:
data2024_clean = data2024_clean.drop(['W','L','ERA','ERA+','WHIP','G.1','GS','SV','IP','H.1','HR.1','BB.1','SO'], axis = 1)
data2024_clean.head()

Rk            Name   YoB % of Ballots  HOFm  HOFs  Yrs    WAR  WAR7  JAWS  \
0   1     Todd Helton   6th        72.2%   175    59   17   61.8  46.6  54.2   
2   3    Andruw Jones   7th        58.1%   109    34   17   62.7  46.4  54.6   
3   4  Gary Sheffield  10th        55.0%   158    61   22   60.5  38.0  49.3   
4   5  Carlos Beltrán   2nd        46.5%   126    52   20   70.1  44.4  57.3   
5   6  Alex Rodriguez   3rd        35.7%   390    77   22  117.5  64.3  90.9   

   ...   HR   RBI   SB    BB     BA    OBP    SLG    OPS  OPS+   Pos Summary  
0  ...  369  1406   37  1335  0.316  0.414  0.539  0.953   133       *3H/7D9  
2  ...  434  1289  152   891  0.254  0.337  0.486  0.823   111      *89H7D/3  
3  ...  509  1676  253  1475  0.292  0.393  0.514  0.907   140  *9*7*5*D6H/3  
4  ...  435  1587  312  1084  0.279  0.350  0.486  0.837   119    *8*9*DH/73  
5  ...  696  2086  329  1338  0.295  0.380  0.550  0.930   140      *6*5DH/3  

[5 rows x 25 columns]

In [19]:
data2024_clean.columns

Index(['Rk', 'Name', 'YoB', '% of Ballots', 'HOFm', 'HOFs', 'Yrs', 'WAR',
       'WAR7', 'JAWS', 'Jpos', 'G', 'AB', 'R', 'H', 'HR', 'RBI', 'SB', 'BB',
       'BA', 'OBP', 'SLG', 'OPS', 'OPS+', 'Pos Summary'],
      dtype='object')

In [20]:
data2024_clean = data2024_clean.drop(['Rk',
                                      'Name',
                                      '% of Ballots',
                                      'Pos\xa0Summary'],
                                     axis = 1)

In [24]:
data2024_clean.columns

Index(['YoB', 'HOFm', 'HOFs', 'Yrs', 'WAR', 'WAR7', 'JAWS', 'Jpos', 'G', 'AB',
       'R', 'H', 'HR', 'RBI', 'SB', 'BB', 'BA', 'OBP', 'SLG', 'OPS', 'OPS+'],
      dtype='object')

In [26]:
data2024_clean.head()

YoB  HOFm  HOFs  Yrs    WAR  WAR7  JAWS  Jpos     G     AB  ...     H  \
0   6th   175    59   17   61.8  46.6  54.2  53.4  2247   7962  ...  2519   
2   7th   109    34   17   62.7  46.4  54.6  58.1  2196   7599  ...  1933   
3  10th   158    61   22   60.5  38.0  49.3  56.7  2576   9217  ...  2689   
4   2nd   126    52   20   70.1  44.4  57.3  58.1  2586   9768  ...  2725   
5   3rd   390    77   22  117.5  64.3  90.9  55.4  2784  10566  ...  3115   

    HR   RBI   SB    BB     BA    OBP    SLG    OPS  OPS+  
0  369  1406   37  1335  0.316  0.414  0.539  0.953   133  
2  434  1289  152   891  0.254  0.337  0.486  0.823   111  
3  509  1676  253  1475  0.292  0.393  0.514  0.907   140  
4  435  1587  312  1084  0.279  0.350  0.486  0.837   119  
5  696  2086  329  1338  0.295  0.380  0.550  0.930   140  

[5 rows x 21 columns]

In [27]:
data2024_clean['YoB'] = data2024_clean['YoB'].str.replace(r'\D', '', regex=True).astype(int)

In [28]:
data2024_clean.head()

YoB  HOFm  HOFs  Yrs    WAR  WAR7  JAWS  Jpos     G     AB  ...     H   HR  \
0    6   175    59   17   61.8  46.6  54.2  53.4  2247   7962  ...  2519  369   
2    7   109    34   17   62.7  46.4  54.6  58.1  2196   7599  ...  1933  434   
3   10   158    61   22   60.5  38.0  49.3  56.7  2576   9217  ...  2689  509   
4    2   126    52   20   70.1  44.4  57.3  58.1  2586   9768  ...  2725  435   
5    3   390    77   22  117.5  64.3  90.9  55.4  2784  10566  ...  3115  696   

    RBI   SB    BB     BA    OBP    SLG    OPS  OPS+  
0  1406   37  1335  0.316  0.414  0.539  0.953   133  
2  1289  152   891  0.254  0.337  0.486  0.823   111  
3  1676  253  1475  0.292  0.393  0.514  0.907   140  
4  1587  312  1084  0.279  0.350  0.486  0.837   119  
5  2086  329  1338  0.295  0.380  0.550  0.930   140  

[5 rows x 21 columns]

In [29]:
model.predict(data2024_clean)

array([ 0.53266895,  0.37306976,  0.42710188,  0.7290012 ,  0.33712792,
        0.32611984,  0.38420206,  0.10610897,  0.17935124,  0.06361547,
        0.95821184,  0.02612188,  0.04905767,  0.024618  ,  0.03588759,
        0.0575914 ,  0.0281069 ,  0.01471563,  0.01096614, -0.00173152],
      dtype=float32)

In [30]:
data2024_clean

YoB  HOFm  HOFs  Yrs    WAR  WAR7  JAWS  Jpos     G     AB  ...     H  \
0     6   175    59   17   61.8  46.6  54.2  53.4  2247   7962  ...  2519   
2     7   109    34   17   62.7  46.4  54.6  58.1  2196   7599  ...  1933   
3    10   158    61   22   60.5  38.0  49.3  56.7  2576   9217  ...  2689   
4     2   126    52   20   70.1  44.4  57.3  58.1  2586   9768  ...  2725   
5     3   390    77   22  117.5  64.3  90.9  55.4  2784  10566  ...  3115   
6     8   226    69   19   69.3  39.9  54.6  53.4  2302   8244  ...  2574   
7     7   120    42   24   45.6  26.8  36.2  55.4  2968  10586  ...  2877   
9     5    95    54   18   60.2  41.6  50.9  56.7  2425   8480  ...  2470   
10    3   121    42   17   47.6  32.7  40.1  55.4  2275   9294  ...  2455   
13    4    58    34   19   50.7  30.8  40.7  58.1  2372   8857  ...  2452   
14    1   163    55   21   93.5  48.7  71.1  55.8  2933  11068  ...  3166   
15    1    92    41   15   55.2  39.0  47.1  44.2  1858   6930  ...  2123   
16    1    94    36   16   64.5  49.3  56.9  57.0  1937   6857  ...  1885   
17    1    74    36   14   49.2  39.5  44.3  55.8  1585   5998  ...  1777   
19    1   110    41   15   44.5  34.3  39.4  53.4  1903   7009  ...  2096   
20    1    90    31   15   43.5  34.6  39.1  53.4  1929   7139  ...  2050   
21    1    78    26   15   36.7  38.2  37.5  56.7  1798   6051  ...  1496   
22    1    76    36   16   37.5  29.3  33.4  55.4  1877   7552  ...  2138   
23    1    63    37   16   32.0  29.0  30.5  44.2  1973   7297  ...  2153   
25    1    68    28   17   28.4  24.8  26.6  57.0  1902   7378  ...  2029   

     HR   RBI   SB    BB     BA    OBP    SLG    OPS  OPS+  
0   369  1406   37  1335  0.316  0.414  0.539  0.953   133  
2   434  1289  152   891  0.254  0.337  0.486  0.823   111  
3   509  1676  253  1475  0.292  0.393  0.514  0.907   140  
4   435  1587  312  1084  0.279  0.350  0.486  0.837   119  
5   696  2086  329  1338  0.295  0.380  0.550  0.930   140  
6   555  1831   38  1329  0.312  0.411  0.585  0.996   154  
7    80   951  404  1028  0.272  0.336  0.352  0.688    82  
9   288  1363  400  1476  0.291  0.395  0.475  0.870   128  
10  231   936  470   813  0.264  0.324  0.418  0.743    95  
13  353  1391  195   661  0.277  0.331  0.461  0.793   110  
14  477  1707  121   848  0.286  0.339  0.480  0.819   116  
15  143   923   52   939  0.306  0.388  0.439  0.827   124  
16  259  1025  154   724  0.275  0.358  0.465  0.823   117  
17  242   970  196   762  0.296  0.376  0.491  0.867   133  
19  316  1220  108   802  0.299  0.379  0.510  0.889   132  
20  317  1202    6   782  0.287  0.358  0.485  0.843   129  
21  344   975   70  1032  0.247  0.361  0.475  0.836   124  
22  145   719  517   589  0.283  0.334  0.427  0.761   103  
23  246  1178    7   730  0.295  0.360  0.455  0.815   118  
25  211   951  209   420  0.275  0.320  0.420  0.740    95  

[20 rows x 21 columns]

Preditions:
- Todd Helton: 53%
- Andruw Jones: 37%
- Gary Sheffield: 43%
- Carlos Beltran: 73%
- Alex Rodriguez: 34%
- Manny Ramirez: 33%
- Omar Vizquel: 38%
- Bobby Abreu: 11%
- Jimmy Rollins: 18%
- Torii Hunter: 6%
- Adrian Beltre: 96%
- Joe Mauer: 3%
- Chase Utley: 5%
- David Wright: 2%
- Matt Holliday: 4%
- Adrian Gonzalez: 6%
- Jose Bautista: 3%
- Jose Reyes: 1%
- Victor Martinez: 1%
- Brandon Phillips: 0%